Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -68,6 +68,22 @@ public struct DiagnosticsFormatter {
     var isFreeOfAnnotations: Bool {
       return diagnostics.isEmpty && suffixText.isEmpty
     }
+
+    /// Converts a UTF-8 column index into an index that considers each character as a single column, not each UTF-8
+    /// byte.
+    ///
+    /// For example the 👨‍👩‍👧‍👦 character is considered as a single character, not 25 bytes.
+    ///
+    /// Both the input and the output column are 1-based.
+    func characterColumn(ofUtf8Column utf8Column: Int) -> Int {
```

Comment:
I assume you don't think this should be part of the location converter?

Reply:
No, I think it’s quite specific to this use case.